x/clp/keeper/msg_server.go

```diff
@@ -368,47 +368,37 @@ func (k msgServer) Swap(goCtx context.Context, msg *types.MsgSwap) (*types.MsgSw
 		return nil, tokenregistrytypes.ErrPermissionDenied
 	}
 
-	if k.checkSwapPermission(ctx, types.NewAsset(sAsset.Denom), types.SwapPermission_DISABLE_SELL) {
-		return nil, types.ErrNotAllowedToSellAsset
-	}
-	if k.checkSwapPermission(ctx, types.NewAsset(rAsset.Denom), types.SwapPermission_DISABLE_BUY) {
-		return nil, types.ErrNotAllowedToBuyAsset
-	}
+	pmtpCurrentRunningRate := k.GetPmtpRateParams(ctx).PmtpCurrentRunningRate
 
 	liquidityProtectionParams := k.GetLiquidityProtectionParams(ctx)
-	precision := sdk.NewDec(10).Power(uint64(18))
 	maxRowanLiquidityThreshold := liquidityProtectionParams.MaxRowanLiquidityThreshold
 	maxRowanLiquidityThresholdAsset := liquidityProtectionParams.MaxRowanLiquidityThresholdAsset
 	currentRowanLiquidityThreshold := k.GetLiquidityProtectionRateParams(ctx).CurrentRowanLiquidityThreshold
 	var (
-		maxRowanLiquidityThresholdAssetNativePrice sdk.Dec
+		sentValue sdk.Uint
 	)
 
-	// if max rowan liquidity threshold set
-	if maxRowanLiquidityThreshold.GT(sdk.ZeroDec()) {
-		// if asset is rowan then threshold defined in rowan
+	// if liquidity protection is active and selling rowan
+	if liquidityProtectionParams.IsActive && strings.EqualFold(sAsset.Denom, types.NativeSymbol) {
 		if strings.EqualFold(maxRowanLiquidityThresholdAsset, types.NativeSymbol) {
-			maxRowanLiquidityThresholdAssetNativePrice = sdk.MustNewDecFromStr("1.0")
+			sentValue = msg.SentAmount
 		} else {
-			// otherwise retrieve asset pool to retrieve native swap price
-			maxRowanLiquidityThresholdAssetPool, err := k.GetPool(ctx, maxRowanLiquidityThresholdAsset)
+			pool, err := k.GetPool(ctx, maxRowanLiquidityThresholdAsset)
 			if err != nil {
 				return nil, types.ErrMaxRowanLiquidityThresholdAssetPoolDoesNotExist
 			}
-			maxRowanLiquidityThresholdAssetNativePrice = *maxRowanLiquidityThresholdAssetPool.SwapPriceNative
-		}
+			sentValue, err = CalcRowanValue(&pool, pmtpCurrentRunningRate, msg.SentAmount)
```

Comment:
The reason for not using the SwapPriceNative is that it doesn't use the base denominations. I also took the opportunity to change to using the spot price  = X/Y instead of XY/(1+X)**2 as used in the SwapPriceNative . It's not clear whether we should calculate the spot price for every pool on every block or simply calculate the spot price when needed. I went for the second option.